modules/reactor/test/tests/meshgenerators/polygon_concentric_circle_mesh_generator/tests

```diff
@@ -85,171 +126,319 @@
+      expect_err = 'This parameter must be strictly ascending.'
+      detail = "duct_sizes is not strictly ascending."
+    []
+    [err_undersize_duct_sizes]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/ring_radii="2 3" Mesh/gen/ring_intervals="2 2"
+                  Mesh/gen/duct_sizes="2.5 4.0" Mesh/gen/duct_intervals="2 2"
+                  --mesh-only "poly_2d_err.e"'
+      expect_err = 'This parameter must be positive and ensures no overlapping with rings.'
+      detail = "duct_sizes is non-positive or is low enough to be overlapped with rings."
+    []
+    [err_oversize_duct_sizes]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/duct_sizes="3.5 8.0" Mesh/gen/duct_intervals="2 2"
+                  --mesh-only "poly_2d_err.e"'
+      expect_err = 'This parameter must ensure that ducts are smaller than the polygon size.'
+      detail = "duct_sizes is bigger than the polygon."
+    []
+    [err_overlapped_background_boundary]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/background_inner_boundary_layer_width=5.0
+                  Mesh/gen/background_outer_boundary_layer_width=5.0
+                  --mesh-only "poly_2d_err.e"'
+      expect_err = 'must be less then the minimum thickness of the background region.'
+      detail = "boundary layers in background region of a hexagonal cell are overlapped."
+    []
+    [err_ring_boundary_layer_param_size]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/background_block_ids="30" Mesh/gen/background_block_names="back"
+                  Mesh/gen/ring_radii="2 3" Mesh/gen/ring_intervals="2 2"
+                  Mesh/gen/ring_inner_boundary_layer_widths="0.6"
+                  --mesh-only "poly_2d_err.e"'
+      expect_err = 'The inner and outer ring boundary layer parameters must have the same sizes as ring_radii.'
+      detail = "inner and outer ring boundary layer parameters have inconsistent size."
+    []
+    [err_ring_inner_boundary_layer_zero_interval]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/background_block_ids="30" Mesh/gen/background_block_names="back"
+                  Mesh/gen/ring_radii="2 3" Mesh/gen/ring_intervals="2 2"
+                  Mesh/gen/ring_inner_boundary_layer_widths="0.0 0.3"
+                  --mesh-only "poly_2d_err.e"'
+      expect_err = 'Ring inner boundary layer must have non-zero interval if its thickness is not zero.'
+      detail = "ring's inner boundary layer has zero interval but non-zero thickness."
+    []
+    [err_ring_inner_boundary_layer_nonzero_interval]
+      type = 'RunException'
+      input = 'poly_2d.i'
+      cli_args = 'Mesh/gen/background_block_ids="30" Mesh/gen/background_block_names="back"
+                  Mesh/gen/ring_radii="2 3" Mesh/gen/ring_intervals="2 2"
+                  Mesh/gen/ring_inner_boundary_layer_widths="0.0 0.0"
```

Comment:
why do you allow a width of 0?
seems like this could be forbidden with a rangeCheckedParam

Reply:
Allowing width of 0 makes it possible to add boundary layer to some blocks and leave the other blocks free of boundary layers.